src/Components/WebAssembly/Build/src/Tasks/GenerateBlazorBootJson.cs

```diff
@@ -99,21 +99,16 @@ internal void WriteBootJson(Stream output, string entryAssemblyName)
             serializer.WriteObject(writer, result);
         }
 
-        private static string GetResourceFileRelativePath(ITaskItem item)
+        private static string GetResourceName(ITaskItem item)
         {
-            // The build targets use RelativeOutputPath in the case of satellite assemblies, which
-            // will have relative paths like "fr\\SomeAssembly.resources.dll". If RelativeOutputPath
-            // is specified, we want to use all of it.
-            var outputPath = item.GetMetadata("RelativeOutputPath");
+            var name = item.GetMetadata("BootManifestResourceName");
 
-            if (string.IsNullOrEmpty(outputPath))
+            if (string.IsNullOrEmpty(name))
             {
-                // If RelativeOutputPath was not specified, we assume the item will be placed at the
-                // root of whatever directory is used for its resource type (e.g., assemblies go in _bin)
-                outputPath = Path.GetFileName(item.GetMetadata("TargetOutputPath"));
+                throw new Exception($"No BootManifestResourceName was specified for item '{item.ItemSpec}'");
```

Comment:
This should never happen in user scenarios because it's only our code that supplies these items, and we do always supply `BootManifestResourceName` metadata. This exception exists only to help us discover any of our own bugs.